usr/sbin/p11sak/p11sak.c

```diff
@@ -4682,9 +4712,14 @@ static CK_RV iterate_objects(const struct p11sak_objtype *objtype,
         data.private = private;
         data.rc = CKR_OK;
 
+#if defined(_AIX)
+        global_private = &data;
+        qsort(matched_objs, num_matched_objs, sizeof(CK_OBJECT_HANDLE),
```

Comment:
I wonder if the following would produce better readable code:

Instead of ifdefing the iterate_compare() function so that its paraeters are different on AIX, how about the followng:

Leave iterate_compare as is, but add a new AIX specific iterate compare function as follows:

```
static int iterate_compare(const void *a, const void *b, void *private)
{
...
}

#if defined(_AIX)
static int iterate_compare_aix(const void *a, const void *b)
{
    return iterate_compare(a, b, global_private);
}
#endif

```

and then

```
#if defined(_AIX)
        global_private = &data;
        qsort(matched_objs, num_matched_objs, sizeof(CK_OBJECT_HANDLE),
                iterate_compare_aix);   <---use iterate_compare_aix here
#else
        qsort_r(matched_objs, num_matched_objs, sizeof(CK_OBJECT_HANDLE),
                iterate_compare, &data);
#endif
```


You can then also leave the `compare_obj` function inside `struct p11sak_iterate_compare_data` as is, and also the `p11sak_list_obj_compare` function can stay as is (with the private argument).  No need to do the globale_private game again there.